sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -58,7 +58,11 @@ trait BroadcastExchangeLike extends Exchange {
    * For registering callbacks on `relationFuture`.
    * Note that calling this method may not start the execution of broadcast job.
    */
-  def completionFuture: scala.concurrent.Future[broadcast.Broadcast[Any]]
+  final def completionFuture: scala.concurrent.Future[broadcast.Broadcast[Any]] = executeQuery {
```

Comment:
This is kind of the AQE version of "execute", as AQE won't call `execute` of shuffle/broadcast.